src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs

```diff
@@ -9502,5 +10167,784 @@ .maxstack  3
+
+            Assert.Equal("T.M01", node.ToString());
+            VerifyOperationTreeForNode(compilation1, model, node,
+// PROTOTYPE(StaticAbstractMembersInInterfaces): It feels like the "T" qualifier is important for this invocation, but it is not 
+//                                               reflected in the IOperation tree. Should we change the shape of the tree in order
+//                                               to expose this information? 
+@"
+IMethodReferenceOperation: void I1.M01() (IsVirtual) (Static) (OperationKind.MethodReference, Type: null) (Syntax: 'T.M01')
+  Instance Receiver: 
+    null
+");
+        }
+
+        [Fact]
+        public void ConsumeAbstractStaticMethod_ConversionFunctionPointer_04()
+        {
+            var source1 =
+@"
+public interface I1
+{
+    abstract static void M01();
+}
+";
+            var source2 =
+@"
+unsafe class Test
+{
+    static void M02<T>() where T : I1
+    {
+        _ = (delegate*<void>)&T.M01;
+    }
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.NetCoreApp);
+
+            var compilation2 = CreateCompilation(source2, options: TestOptions.DebugDll.WithAllowUnsafe(true),
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.DesktopLatestExtended,
+                                                 references: new[] { compilation1.ToMetadataReference() });
+
+            compilation2.VerifyDiagnostics(
+                // (6,13): error CS9100: Target runtime doesn't support static abstract members in interfaces.
+                //         _ = (delegate*<void>)&T.M01;
+                Diagnostic(ErrorCode.ERR_RuntimeDoesNotSupportStaticAbstractMembersInInterfaces, "(delegate*<void>)&T.M01").WithLocation(6, 13)
+                );
+
+            var compilation3 = CreateCompilation(source2 + source1, options: TestOptions.DebugDll.WithAllowUnsafe(true),
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.DesktopLatestExtended);
+
+            compilation3.VerifyDiagnostics(
+                // (12,26): error CS9100: Target runtime doesn't support static abstract members in interfaces.
+                //     abstract static void M01();
+                Diagnostic(ErrorCode.ERR_RuntimeDoesNotSupportStaticAbstractMembersInInterfaces, "M01").WithLocation(12, 26)
+                );
+        }
+
+        [Fact]
+        public void ConsumeAbstractStaticMethod_ConversionToFunctionPointer_06()
+        {
+            var source1 =
+@"
+public interface I1
+{
+    abstract static void M01();
```

Comment:
Doesn't need to be in this PR, but we likely need to have some tests around `UnmanagedCallersOnly`.

Reply:
> Doesn't need to be in this PR, but we likely need to have some tests around UnmanagedCallersOnly.

What would be unique about this scenario by comparison to consumprion of a regular static method?

Reply:
> What would be unique about this scenario by comparison to consumprion of a regular static method?

We don't support `UnmanagedCallersOnly` on instance methods at all, nor do we support taking a pointer to instance methods. We probably shouldn't support them on abstract statics either, or we should make sure that users are forced to apply the attribute to implementations of an abstract static member in a way that matches up.

Reply:
> We don't support UnmanagedCallersOnly on instance methods at all, nor do we support taking a pointer to instance methods. We probably shouldn't support them on abstract statics either, or we should make sure that users are forced to apply the attribute to implementations of an abstract static member in a way that matches up.

Recorded in [https://github.com/dotnet/roslyn/issues/52221](https://github.com/dotnet/roslyn/issues/52221)